### PR TITLE
Permissions

### PR DESCRIPTION
**Pull request by appellation**
**Please describe the changes this PR makes and why it should be merged:**
- not broken, unlike #2128 
- less complicated than #2126 

**Semantic versioning classification:**  
- [ ] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by bdistin** [comment]
This is breaking for no reason, and returns an inconsistent type that isn't what you show in jsdocs...

**Comment by ZLima12** [comment]
I don't know if I agree with this being less complicated than my solution.

**Comment by appellation** [comment]
Return type is a derp. As for complexity and breaking changes, I see no reason why a simple bitwise operation shouldn't be better than loops and filters; return type is just a side effect of simplifying the operation.

**Comment by ZLima12** [comment]
Idk, this still doesn't seem to work. Try calling `(new DiscordJS.Permissions(40)).missing(new DiscordJS.Permissions(86016))`

**Comment by ZLima12** [comment]
Still doesn't work, try `(new DiscordJS.Permissions(4096)).missing(new DiscordJS.Permissions(16384))`. It says that `4096` is missing, although it's the only permission present. 🤔

File: src/util/Permissions.js [hunk 1]
```diff
@@ -32,12 +32,16 @@ class Permissions {
...
    * Gets all given permissions that are missing from the bitfield.
-   * @param {PermissionResolvable[]} permissions Permissions to check for
+   * @param {PermissionResolvable[]|PermissionResolvable} permissions Permissions to check for
    * @param {boolean} [checkAdmin=true] Whether to allow the administrator permission to override
-   * @returns {PermissionResolvable[]}
+   * @returns {Permissions}
    */
   missing(permissions, checkAdmin = true) {
-    return permissions.filter(p => !this.has(p, checkAdmin));
+    permissions = new this.constructor(~this.constructor.resolve(permissions) & this.bitfield);
```
Comment (ZLima12) [inline src/util/Permissions.js]: You're negating the wrong bitfield, it should be `this.constructor.resolve(permissions) & ~this.bitfield`

**Comment by appellation** [comment]
As I understand it, this method is intended to determine the permissions that the instance is missing from the permissions provided.  My understanding is supported by the fact that the current method does exactly that.  Constructing an instance with only permissions for `SEND_TTS_MESSAGES` (4096) would therefore be missing permissions for `EMBED_LINKS` (16384).

**Comment by ZLima12** [comment]
That's correct, but your implementation doesn't do that. `(new DiscordJS.Permissions(4096)).missing(new DiscordJS.Permissions(16384))` returns an instance of `Permissions` with a bitfield value of 4096, or rather "SEND_TTS_MESSAGES".

**Pull request closed without merging**